bftengine/src/bftengine/ControlStateManager.cpp

```diff
@@ -29,37 +29,60 @@ void ControlStateManager::setStopAtNextCheckpoint(int64_t currentSeqNum) {
   uint64_t seq_num_to_stop_at = (currentSeqNum + 2 * checkpointWindowSize);
   seq_num_to_stop_at = seq_num_to_stop_at - (seq_num_to_stop_at % checkpointWindowSize);
   std::ostringstream outStream;
-  controlStateMessages::StopAtNextCheckpointMessage msg(seq_num_to_stop_at);
-  concord::serialize::Serializable::serialize(outStream, msg);
+  page_.seqNumToStopAt_ = seq_num_to_stop_at;
+  concord::serialize::Serializable::serialize(outStream, page_);
   auto data = outStream.str();
-  state_transfer_->saveReservedPage(getUpdateReservedPageIndex(), data.size(), data.data());
+  state_transfer_->saveReservedPage(resPageOffset(), data.size(), data.data());
 }
 
 std::optional<int64_t> ControlStateManager::getCheckpointToStopAt() {
   if (!enabled_) return {};
-  if (!state_transfer_->loadReservedPage(getUpdateReservedPageIndex(), sizeOfReservedPage_, scratchPage_.data())) {
+  if (page_.seqNumToStopAt_ != 0) return page_.seqNumToStopAt_;
+  if (!state_transfer_->loadReservedPage(resPageOffset(), sizeOfReservedPage_, scratchPage_.data())) {
     return {};
   }
   std::istringstream inStream;
   inStream.str(scratchPage_);
-  controlStateMessages::StopAtNextCheckpointMessage msg;
-  try {
-    concord::serialize::Serializable::deserialize(inStream, msg);
-  } catch (std::exception& e) {
-    LOG_WARN(GL, e.what());
-    return {};
-  }
-  if (msg.seqNumToStopAt_ < 0) {
+  concord::serialize::Serializable::deserialize(inStream, page_);
+  if (page_.seqNumToStopAt_ == 0) return {};
+  if (page_.seqNumToStopAt_ < 0) {
```

Comment:
This seems like a fatal error. I left a similar comment on the previous review. Do you agree?

Reply:
Agreed

Reply:
You also need to actually call std::terminate after this log message.

Reply:
fixed